src/components/form/carForm.tsx

```diff
@@ -49,9 +53,8 @@ export function FormFields({register, errors} : FormFieldsProps) {
                 <FormLabel component="legend">Engine Type</FormLabel>
                 <RadioGroup 
                     row
-                    {...register('engine', {required: 'Engine type is required'})}
                     aria-label="engine">
-                    {engineTypes.map((engine) => (<FormControlLabel value={engine} control={<Radio />} label={engine} />))}
+                    {engineTypes.map((engine) => (<FormControlLabel checked={engine == watch('engine')} {...register('engine', {required: 'Engine type is required'})} value={engine} control={<Radio />} label={engine} />))}
                 </RadioGroup>
```

Comment:
Instead of registering each label, canyou try to move register back to the radio group but with few remarks:
1. Try to use `setValue` ([link](https://react-hook-form.com/docs/useform/setvalue)) from react hook form in RadioGroup `onChange`, it should look like ` onChange={(e) => setValue("engine",e.target.value)}`
2. add value to RadioGroup that will use `watch(engine)`